src/utils/logger.js

```
@@ -1,75 +1,71 @@
 import { getSelfScope } from './get-self-scope';
 
-function noop () {}
+const ENABLE_LOGS_DEFAULT = true; // the default setting on lib initialization
 
-const fakeLogger = {
-  trace: noop,
-  debug: noop,
-  log: noop,
-  warn: noop,
-  info: noop,
-  error: noop
-};
-
-let exportedLogger = fakeLogger;
+const ENABLE_TRACE_LOGS = true;
 
-// let lastCallTime;
-// function formatMsgWithTimeInfo(type, msg) {
-//   const now = Date.now();
-//   const diff = lastCallTime ? '+' + (now - lastCallTime) : '0';
-//   lastCallTime = now;
-//   msg = (new Date(now)).toISOString() + ' | [' +  type + '] > ' + msg + ' ( ' + diff + ' ms )';
-//   return msg;
-// }
+const DEBUG_PREFIX = ''; // use this to prefix Hls.js when needed for debugging
+const DEBUG_PREPEND_TIMESTAMP = false; // use this to prepend with timestamp when needed for debugging
 
-function formatMsg (type, msg) {
-  msg = '[' + type + '] > ' + msg;
-  return msg;
-}
+const noop = () => void 0;
+const self = getSelfScope();
 
-const global = getSelfScope();
+const console = self.console;
 
-function consolePrintFn (type) {
-  const func = global.console[type];
-  if (func) {
-    return function (...args) {
-      if (args[0]) {
-        args[0] = formatMsg(type, args[0]);
-      }
+function bindConsole (method, prefix, prependTime) {
+  const logFn = console[method];
+  if (!logFn) {
+    throw new Error('No console support for: ' + method);
```

It would be better to fail silently and simply use the `noop` instead throwing an `Error`.

---

Fine with me.